File: fs/Euclidean_Algorithm.fs
open System

let rec euclidean (a: int) (b: int) =
    match a % b with
        | 0 -> b
        | m -> euclidean b m

let s = Console.ReadLine().Split(' ')
let (a, b) = (int s.[0], int s.[1])

printfn "%d" <| euclidean (max a b) (min a b)

File: scala/Euclidean_Algorithm.scala
import scala.io.StdIn._

object Main {
    def main(args: Array[String]) {
        val ab = readLine().split(" ").map {_.toInt}
        println(euclidean(ab(0), ab(1)))
    }

    def euclidean(a: Int, b: Int): Int = {
        if (a < b) {
            euclidean(b, a)
        }
        else if (a % b == 0) {
            b
        }
        else {
            euclidean(b, a % b)
        }
    }
}
